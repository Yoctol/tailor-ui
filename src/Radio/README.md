# Radio Docs

```js
import { Radio, Label } from '@yoctol/ui';
```

### Try it with `knobs`:

<!-- STORY -->

## API

This component needs using with `Label`.

| Props       | type                                                                              | Default | Description          |
| ----------- | --------------------------------------------------------------------------------- | ------- | -------------------- |
| `disabled`  | Boolean                                                                           | `false` |                      |
| borderColor | [View on docs](https://github.com/jxnblk/styled-system/blob/master/docs/table.md) |         | From `styled-stsyem` |
| Color       | [View on docs](https://github.com/jxnblk/styled-system/blob/master/docs/table.md) |         | From `styled-stsyem` |
