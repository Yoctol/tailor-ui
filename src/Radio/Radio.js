import PropTypes from 'prop-types';
import React from 'react';
import styled, { css } from 'styled-components';

import Space from '../Grid/Space';

import { Consumer } from './RadioContext';

const RadioWrapper = styled.span`
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
`;

const RadioInner = styled.span`
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: ${p => p.theme.borders.base};
  border-radius: 50%;
  border-color: ${p => p.theme.colors.gray[7]};
  background-color: ${p => p.theme.colors.light};
  color: ${p => p.theme.colors.gray[3]};

  ${p => p.theme.transition};

  &::after {
    content: '';
    position: absolute;
    top: 2px;
    left: 2px;
    width: 10px;
    height: 10px;
    border-radius: 50%;
    opacity: 0;
    background-color: ${p => p.theme.colors.primary};
    transform: scale(0);
    ${p => p.theme.transition};
  }
`;

const StyledRadio = styled.input.attrs({
  type: 'radio',
})`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;

  &:checked + ${RadioInner /* sc-selector */} {
    border-color: ${p => p.theme.colors.primary};
  }

  &:checked + ${RadioInner /* sc-selector */}::after {
    opacity: 1;
    transform: scale(1);
  }
`;

const getMarginPosition = p =>
  p.direction === 'horizontal' ? 'margin-left' : 'margin-top';

const RadioLabelBase = styled.label`
  display: inline-flex;
  align-items: center;
  font-size: ${p => p.theme.fontSizes.base};
  cursor: ${p => (p.disabled ? 'not-allowed' : 'pointer')};

  ${p =>
    p.disabled
      ? css`
          opacity: 0.5;
        `
      : css`
          &:hover ${RadioInner /* sc-selector */} {
            border-color: ${p.theme.colors.primary};
          }
        `};
`;

const RadioLabel = styled(RadioLabelBase)`
  /* stylelint-disable-next-line no-descending-specificity */
  & + ${RadioLabelBase /* sc-selector */} {
    ${getMarginPosition /* sc-prop */}: ${p => p.theme.space[2]};
  }
`;

const Radio = ({
  children,
  disabled,
  checked,
  defaultChecked,
  onChange,
  value,
  ...props
}) => (
  <Consumer>
    {({ _onChange, _isChecked, direction }) => (
      <RadioLabel disabled={disabled} direction={direction} {...props}>
        <RadioWrapper>
          <StyledRadio
            disabled={disabled}
            checked={_isChecked ? _isChecked(value) : checked}
            defaultChecked={defaultChecked}
            onChange={event => {
              onChange(event);
              if (_onChange) _onChange(value);
            }}
          />
          <RadioInner />
        </RadioWrapper>
        <Space px="2">{children}</Space>
      </RadioLabel>
    )}
  </Consumer>
);

Radio.propTypes = {
  /**
   * Specifies whether the Radio is selected
   */
  // eslint-disable-next-line react/require-default-props
  checked: PropTypes.bool,
  /**
   * Specifies the initial state: whether or not the Radio is selected.
   */
  // eslint-disable-next-line react/require-default-props
  defaultChecked: PropTypes.bool,
  /**
   * Disable Radio
   */
  disabled: PropTypes.bool,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};

Radio.defaultProps = {
  disabled: false,
  onChange: () => {},
};

export default Radio;
