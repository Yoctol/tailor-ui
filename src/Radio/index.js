import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { borderColor, color, themeGet } from 'styled-system';

import Label from '../Form/Label';
import { controlShadow } from '../utils/shadow';

const StyledRadio = styled.input.attrs({
  type: 'radio',
})`
  position: absolute;
  opacity: 0;

  & + ${Label /* sc-selector */} {
    position: relative;
    padding: 0;
    color: ${themeGet('colors.gray.3')};
    cursor: pointer;
  }

  & + ${Label /* sc-selector */}::before {
    content: '';
    display: inline-block;
    width: 16px;
    height: 16px;
    margin-right: ${themeGet('space.1')};
    border: ${themeGet('borders.default')};
    border-radius: 999px;
    background: white;
    vertical-align: text-bottom;
    ${borderColor};
  }

  & + ${Label /* sc-selector */}::after {
    content: '';
    position: absolute;
    top: 2px;
    left: 2px;
    width: 12px;
    height: 12px;
    border-radius: 50%;
    ${p => p.theme.transition /* sc-declaration */};
  }

  &:focus + ${Label /* sc-selector */}::before {
    ${controlShadow()};
  }

  &:disabled + ${Label /* sc-selector */} {
    opacity: 0.5;
    cursor: auto;
    pointer-events: none;
  }

  &:disabled + ${Label /* sc-selector */}::before {
    opacity: 0.5;
  }

  &:checked + ${Label /* sc-selector */}::after {
    ${color};
  }
`;

const Radio = props => <StyledRadio {...props} />;

Radio.propTypes = {
  /**
   * Specifies whether the checkbox is selected
   */
  checked: PropTypes.bool,
  /**
   * Disable checkbox
   */
  disabled: PropTypes.bool,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
  ...borderColor.propTypes,
  ...color.propTypes,
};

Radio.defaultProps = {
  disabled: false,
  checked: false,
  onChange: () => {},
  // eslint-disable-next-line react/default-props-match-prop-types
  bg: 'primary',
  // eslint-disable-next-line react/default-props-match-prop-types
  borderColor: 'primary',
};

export default Radio;
