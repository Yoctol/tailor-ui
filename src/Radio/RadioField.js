import PropTypes from 'prop-types';
import React from 'react';
import { Value } from 'react-powerplug';

import FormField from '../Form/FormField';
import Hint from '../Form/Hint';
import Label from '../Form/Label';
import Space from '../Grid/Space';

import Radio from '.';

let seed = 0;

const getUuid = () => {
  seed += 1;
  return `yoctol_radio_${Date.now()}_${seed}`;
};

const RadioField = ({
  children,
  success,
  warning,
  error,
  message,
  ...otherProps
}) => {
  const id = getUuid();
  return (
    <FormField success={success} warning={warning} error={error}>
      <Radio id={id} {...otherProps} />
      <Label htmlFor={id}>{children}</Label>
      {(success || warning || error) && <Hint>{message}</Hint>}
    </FormField>
  );
};

RadioField.propTypes = {
  /**
   * Specifies whether the checkbox is selected
   */
  checked: PropTypes.bool,
  /**
   * The label of checkbox field
   */
  children: PropTypes.string.isRequired,
  /**
   * Disable checkbox
   */
  disabled: PropTypes.bool,
  /**
   * Set the checkbox group status to error
   */
  error: PropTypes.bool,
  /**
   * The message will show when success or warning or error is true
   */
  message: PropTypes.string,
  /**
   * Set the checkbox group status to success
   */
  success: PropTypes.bool,
  /**
   * Set the checkbox group status to warning
   */
  warning: PropTypes.bool,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};

RadioField.defaultProps = {
  disabled: false,
  checked: false,
  error: false,
  message: '',
  success: false,
  warning: false,
  onChange: () => {},
};

export const RadioFieldGroup = ({
  label: groupLabel,
  value: groupValue,
  initialValue,
  options,
  onChange,
  success,
  warning,
  error,
  message,
  ...otherProps
}) => (
  <Value initial={initialValue} onChange={onChange}>
    {({ value: checkedValue, set }) => (
      <FormField success={success} warning={warning} error={error}>
        {groupLabel && <Label>{groupLabel}</Label>}
        {options.map(({ label, value, disabled = false }) => {
          const id = getUuid();
          return (
            <Space mt={2} mb={1} key={label}>
              <Radio
                id={id}
                disabled={disabled}
                checked={
                  groupValue ? value === groupValue : value === checkedValue
                }
                onChange={() => (groupValue ? onChange(value) : set(value))}
                {...otherProps}
              />
              <Label htmlFor={id}>{label}</Label>
            </Space>
          );
        })}
        {(success || warning || error) && <Hint>{message}</Hint>}
      </FormField>
    )}
  </Value>
);

RadioFieldGroup.displayName = 'RadioField.Group';

RadioFieldGroup.propTypes = {
  /**
   * Set the checkbox field group status to error
   */
  error: PropTypes.bool,
  /**
   * To set the initial values
   */
  initialValue: PropTypes.string,
  /**
   * The label of checkbox field group
   */
  label: PropTypes.string,
  /**
   * The message will show when success or warning or error is true
   */
  message: PropTypes.string,
  /**
   * Specifies options
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
      disabled: PropTypes.bool,
    })
  ).isRequired,
  /**
   * Set the checkbox field group status to success
   */
  success: PropTypes.bool,
  /**
   * Used for setting the currently selected value
   */
  value: PropTypes.string,
  /**
   * Set the checkbox field group status to warning
   */
  warning: PropTypes.bool,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};
RadioFieldGroup.defaultProps = {
  label: null,
  error: false,
  message: '',
  success: false,
  warning: false,
  value: null,
  initialValue: null,
  onChange: () => {},
};

RadioField.Group = RadioFieldGroup;

export default RadioField;
