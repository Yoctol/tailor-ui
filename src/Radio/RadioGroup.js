import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { Value } from 'react-powerplug';

import Radio from './Radio';
import { Provider } from './RadioContext';

const RadioGroupFlex = styled.div`
  display: ${p => (p.direction === 'horizontal' ? 'flex' : 'inline-flex')};
  flex-direction: ${p => (p.direction === 'horizontal' ? 'row' : 'column')};
`;

const RadioGroup = ({
  value,
  defaultValue,
  options,
  onChange,
  direction,
  children,
  ...otherProps
}) => (
  <RadioGroupFlex direction={direction}>
    <Value initial={defaultValue} onChange={onChange}>
      {({ value: uncontrolledValue, set }) => (
        <Provider
          value={{
            direction,
            _onChange: _value => (value ? onChange(_value) : set(_value)),
            _isChecked: _value =>
              value ? value === _value : uncontrolledValue === _value,
          }}
        >
          {options
            ? options.map(({ label, value: optionValue, disabled = false }) => (
                <Radio
                  key={label}
                  value={optionValue}
                  disabled={disabled}
                  {...otherProps}
                >
                  {label}
                </Radio>
              ))
            : children}
        </Provider>
      )}
    </Value>
  </RadioGroupFlex>
);

RadioGroup.displayName = 'Radio.Group';

RadioGroup.propTypes = {
  /**
   * For composition render
   */
  children: PropTypes.node,
  /**
   * to specify the direction of the step bar, `horizontal` and `vertical` are currently supported
   */
  defaultValue: PropTypes.string,
  /**
   * To set the initial value
   */
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
  /**
   * Specifies options
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
      disabled: PropTypes.bool,
    })
  ),
  /**
   * Used for setting the currently selected value
   */
  value: PropTypes.string,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};

RadioGroup.defaultProps = {
  children: null,
  direction: 'horizontal',
  value: null,
  defaultValue: '',
  options: null,
  onChange: () => {},
};

export default RadioGroup;
