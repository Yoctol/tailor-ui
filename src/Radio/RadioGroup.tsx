import React, { ReactNode, forwardRef, useCallback, useMemo } from 'react';

import { useFormField } from '../FormField';
import { useOwnValue } from '../hooks';

import { Direction, RadioContext } from './RadioContext';
import { Radio } from './Radio';
import { RadioGroupFlex } from './styles';

export interface RadioGroupProps {
  children?: ReactNode;
  defaultValue?: string;
  direction?: Direction;
  options?: {
    label: ReactNode;
    value: string;
    disabled?: boolean;
  }[];
  value?: string;
  onChange?: (value: string) => void;
}

const RadioGroup = forwardRef<HTMLDivElement, RadioGroupProps>(
  function RadioGroup(
    {
      value,
      defaultValue,
      options = null,
      onChange,
      direction = 'horizontal',
      children,
      ...otherProps
    },
    ref
  ) {
    const [ownValue, setOwnValue] = useOwnValue(
      {
        value,
        defaultValue,
        onChange,
      },
      { fallbackValue: '' }
    );

    const [, , setValue] = useFormField({
      value: ownValue,
    });

    const handleChange = useCallback(
      (newValue) => {
        setOwnValue(newValue);
        setValue(newValue);
      },
      [setOwnValue, setValue]
    );

    const isChecked = useCallback((_value) => ownValue === _value, [ownValue]);

    const child = useMemo(
      () =>
        options
          ? options.map(({ label, value: optionValue, disabled = false }) => (
              <Radio key={optionValue} value={optionValue} disabled={disabled}>
                {label}
              </Radio>
            ))
          : children,
      [children, options]
    );

    return (
      <RadioContext.Provider
        value={{
          direction,
          handleChange,
          isChecked,
        }}
      >
        <RadioGroupFlex ref={ref} direction={direction} {...otherProps}>
          {child}
        </RadioGroupFlex>
      </RadioContext.Provider>
    );
  }
);

export { RadioGroup };
