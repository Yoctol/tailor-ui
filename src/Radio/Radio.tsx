import React, { ChangeEvent, FunctionComponent, useContext } from 'react';
import styled, { css } from 'styled-components';

import tag from 'utils/CleanTag';

import Space from '../Grid/Space';

import RadioContext, { Direction } from './RadioContext';
import RadioGroup from './RadioGroup';

const RadioWrapper = styled.span`
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
`;

const RadioInner = styled.span`
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: ${p => p.theme.borders.base};
  border-radius: 50%;
  border-color: ${p => p.theme.colors.gray400};
  background-color: ${p => p.theme.colors.light};
  color: ${p => p.theme.colors.gray700};

  ${p => p.theme.transition};

  &::after {
    content: '';
    position: absolute;
    top: 2px;
    left: 2px;
    width: 10px;
    height: 10px;
    border-radius: 50%;
    opacity: 0;
    background-color: ${p => p.theme.colors.primary};
    transform: scale(0);
    ${p => p.theme.transition};
  }
`;

const StyledRadio = styled.input.attrs({
  type: 'radio',
})`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;

  &:checked + ${RadioInner /* sc-selector */} {
    border-color: ${p => p.theme.colors.primary};
  }

  &:checked + ${RadioInner /* sc-selector */}::after {
    opacity: 1;
    transform: scale(1);
  }
`;

const getMarginPosition = ({ direction }: { direction: Direction }) =>
  direction === 'horizontal' ? 'margin-left' : 'margin-top';

interface IRadioLabelBaseProps {
  disabled?: boolean;
}

const RadioLabelBase = styled(tag.label)<IRadioLabelBaseProps>`
  display: inline-flex;
  align-items: center;
  font-size: ${p => p.theme.fontSizes.base};
  cursor: ${p => (p.disabled ? 'not-allowed' : 'pointer')};

  ${p =>
    p.disabled
      ? css`
          opacity: 0.5;
        `
      : css`
          &:hover ${RadioInner /* sc-selector */} {
            border-color: ${p.theme.colors.primary};
          }
        `};
`;

const RadioLabel = styled(RadioLabelBase)`
  /* stylelint-disable-next-line no-descending-specificity */
  & + ${RadioLabelBase /* sc-selector */} {
    ${getMarginPosition /* sc-prop */}: ${p => p.theme.space[2]};
  }
`;

export interface IRadioProps {
  /**
   * Specifies whether the Radio is selected
   */
  checked?: boolean;
  /**
   * Specifies the initial state: whether or not the Radio is selected.
   */
  defaultChecked?: boolean;
  /**
   * Disable Radio
   */
  disabled?: boolean;
  /**
   * The callback function that is triggered when the state changes
   */
  onChange?: (event: ChangeEvent) => void;
  value?: string;
}

const Radio: FunctionComponent<IRadioProps> & {
  Group: typeof RadioGroup;
} = ({
  children,
  disabled = false,
  checked,
  defaultChecked,
  onChange,
  value,
  ...props
}) => {
  const { _onChange, _isChecked, direction } = useContext(RadioContext);

  return (
    <RadioLabel disabled={disabled} direction={direction}>
      <RadioWrapper>
        <StyledRadio
          disabled={disabled}
          checked={_isChecked && value ? _isChecked(value) : checked}
          defaultChecked={defaultChecked}
          onChange={(event: ChangeEvent) => {
            if (onChange) {
              onChange(event);
            }
            if (_onChange && value) {
              _onChange(value);
            }
          }}
          {...props}
        />
        <RadioInner />
      </RadioWrapper>
      <Space px="2">{children}</Space>
    </RadioLabel>
  );
};

Radio.Group = RadioGroup;

export default Radio;
