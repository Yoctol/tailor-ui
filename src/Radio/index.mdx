---
name: Radio
menu: Components
route: /components/radio
---

import { PropsTable, Playground } from 'docz';

import Button from '../Button';
import Box from '../Grid/Box';

import Radio from './';
import RadioGroup from './RadioGroup';

# Radio

Radio.

## When To Use

- Used to select a single state in multiple options.
- The difference between Select is that Radio is visible to user and can facilitate the comparison of choice, which makes there shouldn't be too many of them.

## Examples

```js
import { Radio } from 'tailor-ui';
```

### Radio

#### Basic

<Playground>
  <Radio>Radio</Radio>
  <br />
  <Radio defaultChecked>Radio</Radio>
</Playground>

#### Disabled

<Playground>
  <Radio checked disabled>
    Radio
  </Radio>
  <br />
  <Radio disabled>Disabled</Radio>
</Playground>

### Radio.Group

#### Basic

<Playground>
  {() => {
    const [value, setValue] = React.useState('radio-a');

    return (
      <>
        <Radio.Group
          options={[
            { label: 'Radio A', value: 'radio-a' },
            { label: 'Radio B', value: 'radio-b' },
            { label: 'Radio C', value: 'radio-c', disabled: true },
            { label: 'Radio D', value: 'radio-d' },
          ]}
          value={value}
          onChange={setValue}
        />
        <br />
        <Radio.Group
          options={[
            { label: 'Radio A', value: 'radio-a' },
            { label: 'Radio B', value: 'radio-b' },
            { label: 'Radio C', value: 'radio-c', disabled: true },
            { label: 'Radio D', value: 'radio-d' },
          ]}
          defaultValue="radio-b"
          onChange={console.log}
        />
      </>
    );

}}

</Playground>

#### Controlled

<Playground>
  {() => {
    const [value, setValue] = React.useState('radio-a');

    return (
      <>
        <Radio.Group
          options={[
            { label: 'Radio A', value: 'radio-a' },
            { label: 'Radio B', value: 'radio-b' },
            { label: 'Radio C', value: 'radio-c', disabled: true },
            { label: 'Radio D', value: 'radio-d' },
          ]}
          value={value}
          onChange={setValue}
        />
        <br />
        <Button onClick={() => set('radio-d')}>Check last one</Button>
      </>
    );

}}

</Playground>

#### Composition render

<Playground>
  {() => {
    const [value, setValue] = React.useState('radio_1');

    return (
      <>
        <Radio.Group value={value} onChange={setValue}>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_1">radio 1</Radio>
          </Box>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_2">radio 2</Radio>
          </Box>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_3">radio 3</Radio>
          </Box>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_4">radio 4</Radio>
          </Box>
        </Radio.Group>
        <br />
        <Radio.Group defaultValue="radio_1" onChange={console.log}>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_1">radio 1</Radio>
          </Box>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_2">radio 2</Radio>
          </Box>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_3">radio 3</Radio>
          </Box>
          <Box bg="gray300" p="2" m="2">
            <Radio value="radio_4">radio 4</Radio>
          </Box>
        </Radio.Group>
      </>
    );

}}

</Playground>

#### Direction

<Playground>
  {() => {
    const [value, setValue] = React.useState('');

    return (
      <Radio.Group
        direction="vertical"
        options={[
          { label: 'radio 1', value: 'radio_1' },
          { label: 'radio 2', value: 'radio_2' },
          { label: 'radio 3', value: 'radio_3' },
          { label: 'radio 4', value: 'radio_4' },
        ]}
        value={value}
        onChange={setValue}
      />
    );

}}

</Playground>

## API

### Radio

<PropsTable of={Radio} />

### Radio.Group

<PropsTable of={RadioGroup} />
