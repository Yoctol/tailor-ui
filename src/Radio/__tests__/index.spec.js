import React from 'react';

import { render } from 'test/test-utils';

import Radio from '..';

describe('Radio', () => {
  it('should render correctly', () => {
    const { container } = render(<Radio>Radio</Radio>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with props disabled', () => {
    const { container } = render(<Radio disabled>Radio</Radio>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
