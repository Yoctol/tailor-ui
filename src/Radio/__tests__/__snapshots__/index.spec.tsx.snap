// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `
.c5 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c4 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 1px solid;
  border-radius: 50%;
  border-color: #bfbfbf;
  background-color: #ffffff;
  color: #373737;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c4::after {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  opacity: 0;
  background-color: #423b63;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c2:checked + .c3 {
  border-color: #423b63;
}

.c2:checked + .c3::after {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  cursor: pointer;
}

.c0:hover .c3 {
  border-color: #423b63;
}

.c0 + .c6 {
  margin-left: 0.5rem;
}

<label
  class="c0"
  direction="horizontal"
>
  <span
    class="c1"
  >
    <input
      class="c2"
      type="radio"
    />
    <span
      class="c3 c4"
    />
  </span>
  <div
    class="c5"
  >
    Radio
  </div>
</label>
`;

exports[`Radio should render with props disabled 1`] = `
.c5 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c4 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 1px solid;
  border-radius: 50%;
  border-color: #bfbfbf;
  background-color: #ffffff;
  color: #373737;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c4::after {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  opacity: 0;
  background-color: #423b63;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c2:checked + .c3 {
  border-color: #423b63;
}

.c2:checked + .c3::after {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6:hover .c3 {
  border-color: #423b63;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  cursor: not-allowed;
  opacity: 0.5;
}

.c0 + .c7 {
  margin-left: 0.5rem;
}

<label
  class="c0"
  direction="horizontal"
>
  <span
    class="c1"
  >
    <input
      class="c2"
      disabled=""
      type="radio"
    />
    <span
      class="c3 c4"
    />
  </span>
  <div
    class="c5"
  >
    Radio
  </div>
</label>
`;
